crates/worker/src/partition/state_machine/command_interpreter/mod.rs

```diff
@@ -734,7 +728,11 @@ where
                 )
                 .await
             }
-            Timer::Invoke(service_id, service_invocation) => {
+            Timer::Invoke(service_id, mut service_invocation) => {
+                // Remove the execution time from the service invocation request
+                service_invocation.execution_time = None;
+
+                // TODO(optimization) we could skip the outbox here and re-propose this to self.
```

Comment:
I think it would make sense to do this "optimization". Otherwise, we are adding an additional replication round for delayed calls which is not strictly necessary. I think we don't need to re-propose to ourselves if the `service_invocation` is targeted to ourselves. Instead, we can directly call `handle_invoke` and be done with it.

Reply:
Technically, this (skipping the outbox for self-targeted messages) could also apply to immediate background calls or calls.

Reply:
> could also apply to immediate background calls or calls.

Immediate you mean execution_time is past now?

Reply:
With immediate I meant non-delayed invocations (`invoke_time == 0`).

Reply:
Isn't this PR already covering it?

Reply:
When getting a journal entry for a call/background call we go through the outbox because we don't know if the current PP owns the invocation or not, this PR did not change that. Perhaps an optimization is worth implementing anyway, indipendently from this PR?